### ISSUE TITLE
Digiline Filter Ejector doesn't eject n Items x in slot m, if there is also a slot < m,  with < n items x.

### ISSUE THREAD

**Issue by FaberUnserzeit** [mt-mods/pipeworks#15]
If there are two or more slots with the same item name, where the first of these slots has less items than the second one, you can only eject the second stack or parts of it, if you prior to this eject the first stack.
To reproduce this, place a chest, a digiline ejector and a chest in a row, connect the ejector to a lua-controller, put 49 cobble into the first slot of the left chest and 99 cobble into another slot. Name the ejector's channel "eject" and run the LUA-controller with the following code:
```
if (event.type=="program") then
   local m =
   {
      exmatch = true,
      name = "default:cobble",
      count = 50,
   }
   digiline_send("eject",m)
end
```
or alternatively:
```
if (event.type=="program") then
   local m = "default:cobble 50"
   digiline_send("eject",m)
end
```
Edit: note: random slotseq sometimes leads to successful ejecting.

**Comment by The4codeblocks** [comment]
I had this issue when coding a withdrawal system for drawer controllers.
A quick fix would be checking how many items were ejected and ejecting the remainder from another slot.

**Comment by SwissalpS** [comment]
> checking how many items were ejected and ejecting the remainder from another slot.

Yes, that's basically how I do it in-game. A digiline detector tube at output to verify if and how much was ejected.

But we will make the ejectors more inteligent, once we find the time to do so :D

**Comment by The4codeblocks** [comment]
> > checking how many items were ejected and ejecting the remainder from another slot.
> 
> Yes, that's basically how I do it in-game. A digiline detector tube at output to verify if and how much was ejected.
> 
> But we will make the ejectors more intelligent, once we find the time to do so :D

I meant for my solution to be applied in the injector code, but this is also a quick fix.


> once we find the time to do so :D

Or we get a contributor to do it :D

**Comment by SwissalpS** [comment]
injectors use a lot of time already. We need some kind of cache to speed things up. Once we have that, this should also be easier to implement.

**Comment by The4codeblocks** [comment]
did some testing, it's a problem with all injectors, not just the digiline one
